### PR TITLE
add json grammar test

### PR DESCRIPTION
**Pull request by JKortmann**

### CODE REVIEW

File: custom_components/test/fuzzing/fuzzer_utils/GrammarFuzzer.py [hunk 1]
```diff
@@ -136,17 +137,17 @@ def fuzz_min_cost(self, grammar: Grammar, start_symbol: Element) -> str:
...
-        given_cost_grammar: Annotated_Grammar,
-        applications: int,
-        max_applications: int,
+            self,
+            head: Element,
+            given_cost_grammar: Annotated_Grammar,
+            applications: int,
+            max_applications: int,
     ) -> str:
         """TODO: Add summary"""
```
Comment (ThorbenCarl) [inline custom_components/test/fuzzing/fuzzer_utils/GrammarFuzzer.py:146]: Some function descriptions are missing.

File: custom_components/test/fuzzing/fuzzer_utils/grammars/grammar_controls_json.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from custom_components.test.fuzzing.fuzzer_utils.GrammarFuzzer import Grammar, GrammarFuzzer
```
Comment (ThorbenCarl) [inline custom_components/test/fuzzing/fuzzer_utils/grammars/grammar_controls_json.py:1]: See no reason why to import the `GrammarFuzzer`, I think to import `Grammar` is sufficient.

File: custom_components/test/fuzzing/fuzzer_utils/grammars/grammar_controls_json.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+grammar_controls_json: Grammar = {
+    "<JSON>": ["{ <Members> }"],
+    "<Members>": ["<ControlsPair>", "<Members>, <ControlsPair>"],
+    "<ControlsPair>": ["\"<String>\": { <ControlsElement> }", "<ControlsPair>, \"<String>\": { <ControlsElement> }"],
+    "<ControlsElement>": ["\"type\": \"<String>\""],
+    "<String>": ["<Char>"],
+    "<Char>": ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u",
```
Comment (ThorbenCarl) [inline custom_components/test/fuzzing/fuzzer_utils/grammars/grammar_controls_json.py:9]: If I understand it correctly, your string always consists of only one char, and not of any number of chars. So there is not even a theoretical probability that the key word `controls` will be created? In the log files I always see only one char.

File: custom_components/test/fuzzing/value_pools/test_vp_on_helpers.py [hunk 4]
```diff
@@ -135,12 +134,10 @@ def test_get_miniserver_type() -> None:
...
 
 
-
 @pytest.mark.timeout(300)
-@pytest.mark.skip(reason="No meaningful dict.")
 def test_get_all() -> None:
     logger.info("Start of get_all() test.")
-    param_set = value_pool_fuzzer.fuzz(["DICT", "STRING"], 2)
+    param_set = value_pool_fuzzer.fuzz(["GRAMMAR_CONTROLS_JSON_COV", "STRING"], 2)
     result = param_runner.run(get_all, param_set)
```
Comment (ThorbenCarl) [inline custom_components/test/fuzzing/value_pools/test_vp_on_helpers.py:141]: As expected, the testcase always fails, the expected strings in the `get_all()` function are never generated. Of course, it is also debatable whether the function `get_all()` in PyLoxone should not intercept this better.

**Review by ThorbenCarl — CHANGES_REQUESTED** [review]
I think it would be cool if we added grammars to the functions `test_get_room_name_from_room_uuid()`, `test_get_cat_name_from_cat_uuid()` and `test_add_room_and_cat_to_value_values()`, then we would have the `helper.py` file completely covered.

File: custom_components/test/fuzzing/fuzzer_utils/grammars/grammar_controls_json.py (hunk 3 above)
Comment (JKortmann) [reply custom_components/test/fuzzing/fuzzer_utils/grammars/grammar_controls_json.py:9]: Totally forgot that in the end. Added the controls key to generate fitting json for the function. Good catch!

File: custom_components/test/fuzzing/value_pools/test_vp_on_helpers.py (hunk 4 above)
Comment (JKortmann) [reply custom_components/test/fuzzing/value_pools/test_vp_on_helpers.py:141]: Good question. We may discuss this in the weekly. For now I think that's how we should test this function given the fuzzing approach.

File: custom_components/test/fuzzing/fuzzer_utils/GrammarFuzzer.py (hunk 1 above)
Comment (ThorbenCarl) [reply custom_components/test/fuzzing/fuzzer_utils/GrammarFuzzer.py]: ref to [7d2de52](https://github.com/ThorbenCarl/PyLoxone/pull/18/commits/7d2de52eafbe877af951ea7f5e2d3affa08d3395) I still found 7 `"""TODO: Add summary"""` 😉

File: custom_components/test/fuzzing/fuzzer_utils/grammars/grammar_controls_json.py (hunk 3 above)
Comment (ThorbenCarl) [reply custom_components/test/fuzzing/fuzzer_utils/grammars/grammar_controls_json.py:9]: I would still change line 8 to: `"<String>": ["<Char>", "<Char><String>"],`.
I think we should test longer strings then only one char.

File: custom_components/test/fuzzing/value_pools/test_vp_on_helpers.py (hunk 4 above)
Comment (ThorbenCarl) [reply custom_components/test/fuzzing/value_pools/test_vp_on_helpers.py:141]: After [7505239](https://github.com/ThorbenCarl/PyLoxone/pull/18/commits/750523981a0fb60d1c638f5f280f27f8aaa67e5a) the test passes.

**Review by ThorbenCarl — CHANGES_REQUESTED** [review]
See two open points above.

**Review by ThorbenCarl — APPROVED** [review]

**Pull request merged**